tests/integration/integration_test_Entity.py

```diff
@@ -433,7 +433,13 @@ def test_create_or_update_project():
     name = str(uuid.uuid4())
 
     project = Project(name, a=1, b=2)
-    syn.store(project)
+    proj_for_cleanup = syn.store(project)
+    schedule_for_cleanup(proj_for_cleanup)
+
+    #TODO: this is here because store() calls _findEntityIdByNameAndParent() in the case that we store() an entity without
+    #TODO: an id, like we do below. _findEntityIdByNameAndParent() uses query() which is eventually consistent but not immediate
+    #TODO: (i.e. we can't immediately query for an Entity ID using projet name and parentId right after creating it)
+    time.sleep(3)
 
```

Comment:
_findEntityIdByNameAndParent should use the new API for that purpose (http://docs.synapse.org/rest/POST/entity/child.html).